sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/coders/AwsCoders.java

```diff
@@ -0,0 +1,169 @@
+      MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())
+          .encode(
+              ImmutableMap.of(
+                  ResponseMetadata.AWS_REQUEST_ID, value != null ? value.getRequestId() : ""),
+              outStream);
+    }
+
+    @Override
+    public ResponseMetadata decode(InputStream inStream) throws CoderException, IOException {
+      return new ResponseMetadata(
+          MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()).decode(inStream));
+    }
+
+    @Override
+    public List<? extends Coder<?>> getCoderArguments() {
+      return ImmutableList.of();
+    }
+
+    @Override
+    public void verifyDeterministic() throws NonDeterministicException {
+      MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()).verifyDeterministic();
+    }
+  }
+
+  private static class FullSdkHttpMetadataCoder extends Coder<SdkHttpMetadata>
+      implements Serializable {
+
+    private static final FullSdkHttpMetadataCoder INSTANCE = new FullSdkHttpMetadataCoder();
+
+    private FullSdkHttpMetadataCoder() {}
+
+    public static FullSdkHttpMetadataCoder of() {
+      return INSTANCE;
+    }
+
+    @Override
+    public void encode(SdkHttpMetadata value, OutputStream outStream)
+        throws CoderException, IOException {
+      VarInt.encode(value != null ? value.getHttpStatusCode() : 0, outStream);
+      MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())
+          .encode(value != null ? value.getHttpHeaders() : ImmutableMap.of(), outStream);
+    }
+
+    @Override
+    public SdkHttpMetadata decode(InputStream inStream) throws CoderException, IOException {
+      final int httpStatusCode = VarInt.decodeInt(inStream);
+      final Map<String, String> httpHeaders =
+          MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()).decode(inStream);
+      HttpResponse httpResponse = new HttpResponse(null, null);
+      httpResponse.setStatusCode(httpStatusCode);
+      httpHeaders.keySet().forEach(k -> httpResponse.addHeader(k, httpHeaders.get(k)));
+      return SdkHttpMetadata.from(httpResponse);
+    }
+
+    @Override
+    public List<? extends Coder<?>> getCoderArguments() {
+      return ImmutableList.of();
+    }
+
+    @Override
+    public void verifyDeterministic() throws NonDeterministicException {
+      VarIntCoder.of().verifyDeterministic();
+      MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()).verifyDeterministic();
+    }
+  }
+
+  private static class MinimalSdkHttpMetadataCoder extends Coder<SdkHttpMetadata>
+      implements Serializable {
+
+    private static final MinimalSdkHttpMetadataCoder INSTANCE = new MinimalSdkHttpMetadataCoder();
+
+    private MinimalSdkHttpMetadataCoder() {}
+
+    public static MinimalSdkHttpMetadataCoder of() {
+      return INSTANCE;
+    }
+
+    @Override
+    public void encode(SdkHttpMetadata value, OutputStream outStream)
+        throws CoderException, IOException {
+      VarInt.encode(value != null ? value.getHttpStatusCode() : 0, outStream);
```

Comment:
Same suggestion for wrapping NullableCoder within `minimalSdkHttpMetadata` and not handling the null case here.